src/modules/phoneNumber/utils.ts

```diff
@@ -0,0 +1,23 @@
+import { PHONE_REGEX_MAPPER } from './data/phoneRegexMapper';
+
+export const detectCountryCodeFromDialCode = (
+  phoneNumber: string | number,
+): string => {
+  for (const countryCode in PHONE_REGEX_MAPPER) {
```

Comment:
Is there any better way to handle this ? ...any idea how libphonenumber handles this ?
This will be traversing across all regex patterns for each country for every function call.

Reply:
libphonenumber-js utilizes metadata compiled from the Google's libphonenumber project, which includes a comprehensive set of phone number-related metadata, such as country codes, number formats, and national dialing rules. 

They have their own algo that not only uses regex but also looks into number of digits and their specific arrangement and matches these with the above mentioned metadata. We can also have somewhat similar thing where not only we use regex but we can also use number of digits and dial codes matching to figure out the country, will need some time for this though.

I also looked into Checkout Codebase they also seem to use Regex checks to figure this out. Screenshot of the same is attached below:

![Uploading Screenshot 2023-11-29 at 12.47.06 AM.png…]()